challenge.rb

```diff
@@ -1,26 +1,32 @@
 
 def capitalize_each_string(input)
   #implement your solution here
+  input.map {|s| s.capitalize}
 end
 
 def fetch_the_dog(input)
   #implement your solution here
+  input.find_all {|s| s.downcase == "dog"}
 end
 
 def no_dogs_allowed(input)
   #implement your solution here
+  input.reject {|s| s.downcase == "dog"}
 end
 
 def count_the_animals(input)
   #implement your solution here
+  input.count
 end
 
 def fetch_the_first_two(input)
   #implement your solution here
+  input.take(2)
 end
 
 def fetch_CD_animals(input)
   #implement your solution here
+  input.select{|s| s.downcase =~ /(c|d)[a-z]+/}
```

Comment:
Cool - this is on the right track.  But this regex will *capture* a 'c' or 'd' which is followed by at least one other letter - doesn't matter what comes before the 'c' or 'd'.  But we just want to *match* (not capture) an expression that *starts with* 'c' or 'd'.  I'd say give it one more shot, or if you really want I can just say what I've got in mind.